cmd/kubectl/kubectlcobra/grouping_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// package kubectlcobra contains cobra commands from kubectl
+package kubectlcobra
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+var testNamespace = "test-grouping-namespace"
+var groupingObjName = "test-grouping-obj"
+var pod1Name = "pod-1"
+var pod2Name = "pod-2"
+var pod3Name = "pod-3"
+
+var groupingObj = corev1.ConfigMap{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: testNamespace,
+		Name:      groupingObjName,
+		Labels: map[string]string{
+			GroupingLabel: "true",
+		},
+	},
+}
+
+var groupingInfo = &resource.Info{
+	Namespace: testNamespace,
+	Name:      groupingObjName,
+	Object:    &groupingObj,
+}
+
+var pod1 = corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: testNamespace,
+		Name:      pod1Name,
+	},
+}
+
+var pod1Info = &resource.Info{
+	Namespace: testNamespace,
+	Name:      pod1Name,
+	Object:    &pod1,
+}
+
+var pod2 = corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: testNamespace,
+		Name:      pod2Name,
+	},
+}
+
+var pod2Info = &resource.Info{
+	Namespace: testNamespace,
+	Name:      pod2Name,
+	Object:    &pod2,
+}
+
+var pod3 = corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: testNamespace,
+		Name:      pod3Name,
+	},
+}
+
+var pod3Info = &resource.Info{
+	Namespace: testNamespace,
+	Name:      pod3Name,
+	Object:    &pod3,
+}
+
+func TestIsGroupingObject(t *testing.T) {
```

Comment:
I find that tests are easier to debug if we name each testcase and use subtests. So something like:
```
testCases := map[string]struct{
  obj runtime.Object
}{
  "name of testcase": {
    obj: ...
  }
}

for tn, tc := range testcases {
  t.Run(tn, func(t *testing.T) {
    <test code>
  }
}
```
This will cause all testcases to be listed as separate tests when they are running and if something fails, it is easy to tell which case failed.

Reply:
This sounds like a good idea. I will implement this test naming in future PR's.